zookeeper-server/src/main/java/org/apache/zookeeper/metrics/SummarySet.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.metrics;
+
+/**
+ * Summaries track the size and number of events.
+ * They are able to publish minumum, maximum, average values, depending on the capabilities of the MetricsProvider.
+ * A SummarySet is a set of {@link Summary}.
+ */
+public interface SummarySet {
+
+     /**
+      * Register a value.
+      * <p>This method is thread safe, The MetricsProvider will take care of synchronization.</p>
+      *
+      * @param key the key to access the Summary for the given key
+      * @param value current value
+      */
+     void add(String key, long value);
```

Comment:
Yes, this definitely covers Set metrics we have. Though I was kind of hoping we could play some tricks with context to make it work without SummarySet.

Reply:
@jhuan31  we can support MetricsContext hierarchy as well.
This approach with "SummarySet" will be very useful with Prometheus, we will make 'key' to "label" in Prometheus API and this will help the creation of queries.

I have such case in another application (not OSS), originally I did not think it would be useful in ZooKeeper (as I did not see your stuff while designing the first version on the API).

So I would keep both of the approaches, in this case the keyed approach works better